index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
```

Comment:
hmm...we're going to need to rethink this here. calling `pinMode` will revert the pin from using the Sonar class you implemented and instead assign it for use with the GPIO class.

I think that we should add this logic to the `pinMode` private method. I think we should create a new (internal only) mode called `PING_READ_MODE` and assign it a value of 98. Then we can call `this[pinMode](pin, PING_READ_MODE)` right here. The `new Sonar()` call will be moved to the pin mode method, although the rest of the logic will remain here. There's one catch though, we'll have to set the mode on the pinInstance to `UNKNOWN_MODE` though inside of the `pinMode` method, since Johnny-Five won't recognize `PING_READ_MODE`.

The sonar instance can be referenced here using `pinInstance.peripheral` to set up the read handler and such.

Let me know if any of this is unclear, I know it's a lot to take in.